Updates based on Product review & usability improvements

emmarogge opened the pull request:
Address Product feedback on notebooks. See details in individual commit messages.

emmarogge committed:
Add table of contents links to Objective section

emmarogge committed:
Strip widget input values before using in command

Strip the widgets' input values before using them to run Terra CLI commands. Also refactor certain methods to use one list instead of multiple lists for parameters.

emmarogge committed:
Respond to Product feedback

- Clarify name and information about Workspace Name field. It is not required, but not providing a value results in ugly UUID. See [bug](https://broadworkbench.atlassian.net/browse/PF-2830) describing divergent behavior of CLI vs UI.
- Add a widget for converting an existing workspace to a data collection, with dropdown list of eligible workspace IDs.
- Add whitespace stripping where appropriate.
- Add hyperlinked Table of Contents section and reorder sections to accommodate new section.

emmarogge committed:
Add message directing users to contact support for group policy constraint assistance

wnojopra commented on first_hour_on_vwb/widget_utils.py:
```diff
@@ -57,6 +57,26 @@ def list_group_members(json_string):
     return html
 
+def list_workspace_ids(json_string):
+    """ Return a list of workspace IDs.
```
In the comment, clarify that the input to this function is the output from terra workspace list

emmarogge replied:
done!

wnojopra commented on first_hour_on_vwb/widget_utils.py:
```diff
@@ -57,6 +57,26 @@ def list_group_members(json_string):
+    """
+    json_data = json.loads(json_string)
+    id_list = []
+    for row in json_data:
+        ws_id = row['id']
+        ws_properties = row['properties']
+        if 'terra-type' in ws_properties:
+            terra_type = ws_properties['terra-type']
+            if terra_type == 'data-collection':
+                continue
+            # Workspaces do not have an explictly set terra-type
+            # but may have user-set properties.
+            else:
+                id_list.append(ws_id)
+        else:
+            id_list.append(ws_id)
+    return id_list
```
This may be cleaner:
```suggestion
    for row in json_data:
        ws_id = row['id']
        ws_properties = row['properties']
        if ws_properties.get('terra-type') == 'data-collection':
            # Workspaces do not have an explictly set terra-type
            # but may have user-set properties.
            continue
        id_list.append(ws_id)
    return id_list
```

emmarogge replied:
done!

wnojopra commented on first_hour_on_vwb/working_with_data_collections.ipynb:
```diff
@@ -43,15 +43,18 @@
-    "1. Create a new data collection from cloud data.\n",
-    "1. Share the data collection with collaborators.\n",
-    "1. Add the data collection as a resource to your workspace.\n",
+    "- [Create a new data collection](#create-new-dc)\n",
```
Is there anything I need to do to get these links to work? When I click on them, nothing happens for me

emmarogge replied:
I used 'href' instead of 'id'. I have now fixed these and tested them.

wnojopra commented on first_hour_on_vwb/working_with_data_collections.ipynb:
```diff
@@ -103,16 +104,17 @@
     "\n",
-    "The process of creating a data collection requires you to specify the following:\n",
+    "Before creating your data collection, consider the following:\n",
     "1. What data do you want to share? What type of resources--Cloud Storage buckets or objects, BigQuery tables, GitHUb repositories--will be made available via this data collection?\n",
```
```suggestion
    "1. What data do you want to share? What type of resources--Cloud Storage buckets or objects, BigQuery tables, GitHub repositories--will be made available via this data collection?\n",
```

emmarogge replied:
fixed!

wnojopra commented on first_hour_on_vwb/working_with_data_collections.ipynb:
```diff
@@ -103,16 +104,17 @@
     "<div class=\"alert alert-block alert-success\">\n",
     "<b>Note:</b> \n",
-    "    If you'd like to restrict access to your data collection to members of a specific group, you'll need to provide the <a href=\"https://et-docs-tests.googleplex.com/docs/reference/glossary/#policy\">group policy constraint</a> at the time of workspace creation. See <a href=\"../creating_a_group.ipynb\">../creating_a_group.ipynb</a> for details on how to create a Verily Workbench group that ...
+    "    If you'd like to restrict access to your data collection to members of a specific group, contact <a href=\"mailto:workbench-support@verily.comworkbench-support@verily.com\">Verily Workbench Support</a> <b>before</b> creating your workspace so that a <a href=\"https://et-docs-tests.googleplex.com/docs/reference/glossary/#policy\">group policy constraint</a> can be applied at the time of w ...
```
I just noticed this still links to et-docs-tests.googleplex.com. Should this be updated?

amygdala replied:
Yes, it should now be: https://terra-docs.api.verily.com/docs/

emmarogge replied:
fixed!

wnojopra commented on first_hour_on_vwb/working_with_data_collections.ipynb:
```diff
@@ -103,16 +104,17 @@
     "<div class=\"alert alert-block alert-success\">\n",
     "<b>Note:</b> \n",
-    "    If you'd like to restrict access to your data collection to members of a specific group, you'll need to provide the <a href=\"https://et-docs-tests.googleplex.com/docs/reference/glossary/#policy\">group policy constraint</a> at the time of workspace creation. See <a href=\"../creating_a_group.ipynb\">../creating_a_group.ipynb</a> for details on how to create a Verily Workbench group that ...
+    "    If you'd like to restrict access to your data collection to members of a specific group, contact <a href=\"mailto:workbench-support@verily.comworkbench-support@verily.com\">Verily Workbench Support</a> <b>before</b> creating your workspace so that a <a href=\"https://et-docs-tests.googleplex.com/docs/reference/glossary/#policy\">group policy constraint</a> can be applied at the time of w ...
```
Is the formatting of the support email address correct? Looks like its pasted twice

emmarogge replied:
done!

wnojopra commented on first_hour_on_vwb/working_with_data_collections.ipynb:
```diff
@@ -122,22 +124,25 @@
+    "- `Workspace Name`: Must be a string. This value is displayed in the Data Collection modal once the workspace is converted to a data collection, so the value should communicate the intended purpose (e.g. `<STUDY_NAME> Data Collection`).<br> While the Verily Workbench UI and this widget require a workspace name to be provided, the CLI does not; if no workspace name is provided to the CLI, a U ...
+    "- `Description`: Must be a string. This description will stay with the workspace after it becomes a data collection. Before you convert a workspace to a data collection, you can update this value in the UI.\n",
+    "- `Workspace ID`: Must be unique and consist only of lowercase letters, numbers and underscores. Provide a workspace ID that suggests something about the contents of the data collection you'd like to create and include the date of its creation, such as `<STUDY_NAME>_<YYMMDD>_dc_ws`. *You cannot change the workspace ID after workspace creation.* \n",
+    "- `Workbench Group Name`: Must be a Verily Workbench group in which you are a member or admin. Group policy constraints for a given workspace must be set at the time of workspace creation, and are necessary but insufficient for granting members of the group workspace access.\n",
```
I don't see an input field in the widget for group

emmarogge replied:
I have removed this content now that I've confirmed with Eng that it's not possible to add a group policy constraint via the CLI, and therefore the group name is not a parameter for this widget.

emmarogge committed:
Respond to reviewer feedback

- Clarify that the input to a function is the output from a specific CLI command.
- Simplify function that lists non-data-collection workspaces.
- Fix Table of Contents in data collections notebook -- s/href/id.
- Fix typo.
- Replace URL with user-facing docs site URL.
- Use correct email address for Verily Support in link.
- Remove reference to group name parameter as it's not possible to add the group policy constraint via CLI.

wnojopra commented on first_hour_on_vwb/working_with_data_collections.ipynb:
```diff
@@ -103,16 +104,17 @@
     "<div class=\"alert alert-block alert-success\">\n",
     "<b>Note:</b> \n",
-    "    If you'd like to restrict access to your data collection to members of a specific group, you'll need to provide the <a href=\"https://et-docs-tests.googleplex.com/docs/reference/glossary/#policy\">group policy constraint</a> at the time of workspace creation. See <a href=\"../creating_a_group.ipynb\">../creating_a_group.ipynb</a> for details on how to create a Verily Workbench group that ...
+    "    If you'd like to restrict access to your data collection to members of a specific group, contact <a href=\"mailto:workbench-support@verily.com\">Verily Workbench Support</a> <b>before</b> creating your workspace so that a <a href=\"https://terra-docs.api.verily.com/docs/https://terra-docs.api.verily.com/docs/reference/glossary/#policy\">group policy constraint</a> can be applied at the t ...
```
1. Is this the right email? On terra.verily.com it says terra-support@verily.com
2. The link to the group policy constraint page is wrong

wnojopra replied:
Scratch the first point. As of last night workbench-support@ is official.

emmarogge replied:
thank you for catching that -- fixed!
